diff --git a/builds/windows/ftsystem.c b/builds/windows/ftsystem.c
index 1ebadd4..47fcb16 100644
--- a/builds/windows/ftsystem.c
+++ b/builds/windows/ftsystem.c
@@ -4,7 +4,7 @@
  *
  *   Windows-specific FreeType low-level system interface (body).
  *
- * Copyright (C) 2021 by
+ * Copyright (C) 2021-2022 by
  * David Turner, Robert Wilhelm, and Werner Lemberg.
  *
  * This file is part of the FreeType project, and may only be used,
@@ -196,7 +196,10 @@
   }
 
 
-#ifdef _WIN32_WCE
+#define UWP_LEGACY
+
+
+#if defined( _WIN32_WCE ) || defined( UWP_LEGACY )
 
   FT_LOCAL_DEF( HANDLE )
   CreateFileA( LPCSTR                lpFileName,
@@ -233,6 +236,10 @@
                         dwFlagsAndAttributes, hTemplateFile );
   }
 
+#endif
+
+#if defined( _WIN32_WCE ) || defined ( _WIN32_WINDOWS ) || \
+    !defined( _WIN32_WINNT ) || _WIN32_WINNT <= 0x0400
 
   FT_LOCAL_DEF( BOOL )
   GetFileSizeEx( HANDLE         hFile,
@@ -248,7 +255,7 @@
       return TRUE;
   }
 
-#endif /* _WIN32_WCE */
+#endif
 
 
   /* documentation is in ftobjs.h */
