diff --git a/libarchive/archive.h b/libarchive/archive.h
index 5b65475..564da5d 100644
--- a/libarchive/archive.h
+++ b/libarchive/archive.h
@@ -725,6 +725,7 @@ __LA_DECL int		 archive_read_free(struct archive *);
 /* Synonym for archive_read_free() for backwards compatibility. */
 __LA_DECL int		 archive_read_finish(struct archive *) __LA_DEPRECATED;
 #endif
+__LA_DECL int		 archive_read_is_end_of_file(struct archive *);
 
 /*-
  * To create an archive:
diff --git a/libarchive/archive_read.c b/libarchive/archive_read.c
index 822c534..596ccaa 100644
--- a/libarchive/archive_read.c
+++ b/libarchive/archive_read.c
@@ -943,6 +943,15 @@ archive_read_data_skip(struct archive *_a)
 	return (r);
 }
 
+int
+archive_read_is_end_of_file(struct archive *_a)
+{
+	struct archive_read *a = (struct archive_read *)_a;
+	if (!a->filter)
+		return (0);
+	return (a->filter->end_of_file);
+}
+
 la_int64_t
 archive_seek_data(struct archive *_a, int64_t offset, int whence)
 {
